### PR TITLE
add callplayaudio

### PR DESCRIPTION
**Pull request by v-maku23**
## Purpose

* ...

## Does this introduce a breaking change?

```
[ ] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

**Review by v-mausha — COMMENTED** [review]
.

File: CallPlayAudio/src/main/java/com/communication/CallPlayAudio/CallPlayTerminate.java [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+import com.azure.communication.callingserver.CallConnection;
+import com.azure.communication.callingserver.CallingServerClient;
+import com.azure.communication.callingserver.CallingServerClientBuilder;
+import com.azure.communication.callingserver.models.CallConnectionState;
+import com.azure.communication.callingserver.models.CancelAllMediaOperationsResult;
+import com.azure.communication.callingserver.models.CreateCallOptions;
+import com.azure.communication.callingserver.models.EventSubscriptionType;
+import com.azure.communication.callingserver.models.MediaType;
+import com.azure.communication.callingserver.models.OperationStatus;
+import com.azure.communication.callingserver.models.PlayAudioOptions;
```
Comment (v-mausha) [inline CallPlayAudio/src/main/java/com/communication/CallPlayAudio/CallPlayTerminate.java:11]: Remove unused imports.

File: CallPlayAudio/src/main/java/com/communication/CallPlayAudio/CallPlayTerminate.java [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+    private final CallConfiguration callConfiguration;
+    private final CallingServerClient callingServerClient;
+    private CallConnection callConnection = null;
+    private CancellationTokenSource reportCancellationTokenSource;
+    private CancellationToken reportCancellationToken;
+    private CompletableFuture<Boolean> callConnectedTask;
+    private CompletableFuture<Boolean> playAudioCompletedTask;
+    private CompletableFuture<Boolean> callTerminatedTask;
+    private CompletableFuture<Boolean> toneReceivedCompleteTask;
+
```
Comment (v-mausha) [inline CallPlayAudio/src/main/java/com/communication/CallPlayAudio/CallPlayTerminate.java:40]: Remove unused variables

File: CallPlayAudio/src/main/java/com/communication/CallPlayAudio/CallPlayTerminate.java [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+                //     }
+                //     return false;
+                // });
+
+                // CompletableFuture<Object> completedTask = CompletableFuture.anyOf(playAudioCompletedTask, maxWait);
+                // if (completedTask.get() != playAudioCompletedTask.get()) {
+                //     Logger.logMessage(Logger.MessageType.INFORMATION, "No response from user in 30 sec, initiating hangup");
+                //     playAudioCompletedTask.complete(false);
+                //     toneReceivedCompleteTask.complete(false);
+                // }
```
Comment (v-mausha) [inline CallPlayAudio/src/main/java/com/communication/CallPlayAudio/CallPlayTerminate.java]: Remove commented code

File: CallPlayAudio/src/main/java/com/communication/CallPlayAudio/CallPlayTerminate.java [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+            Logger.logMessage(Logger.MessageType.INFORMATION, "Cancellation request, Hangup will not be performed");
+            return;
+        }
+
+        Logger.logMessage(Logger.MessageType.INFORMATION, "Performing Hangup operation");
+        Response<Void> response = this.callConnection.hangupWithResponse(null);
+        Logger.logMessage(Logger.MessageType.INFORMATION, "hangupWithResponse -- > " + getResponse(response));
+    }
+
+    private void registerToPlayAudioResultEvent(String operationContext) {
```
Comment (v-mausha) [inline CallPlayAudio/src/main/java/com/communication/CallPlayAudio/CallPlayTerminate.java]: Remove these unused methods registerToPlayAudioResultEvent, getIdentifierKind, and cancelMediaProcessing, and others if any.

File: CallPlayAudio/src/main/java/com/communication/CallPlayAudio/App.java [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+                for (String identity : identities) {
+                    String[] pair = identity.split(",");
+                    if(pair.length == 2 && !pair[0].isEmpty() && !pair[1].isEmpty()) {
+                        tasks.add(() -> {
+                            new CallPlayTerminate(callConfiguration).report(pair[0].trim());
+                            return true;
+                        });
+                    } else {
+                        Logger.logMessage(Logger.MessageType.ERROR, "Malformed destination identitity --> " + identity);            
+                    }
```
Comment (v-mausha) [inline CallPlayAudio/src/main/java/com/communication/CallPlayAudio/App.java:98]: As we don't need the identities, so there should be no need to have the pair logic, the identity should be in the format 
phone1; phone2; phone3; ..............; phoneN
and pass phone number in the report method as argument.

File: CallPlayAudio/src/main/java/com/communication/CallPlayAudio/App.java [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+    }
+
+    private static void runSample(String appBaseUrl) {
+        CallConfiguration callConfiguration = initiateConfiguration(appBaseUrl);
+        ConfigurationManager configurationManager = ConfigurationManager.getInstance();
+        String callplayaudioPairs = configurationManager.getAppSettings("DestinationIdentities");
+
+        Logger.logMessage(Logger.MessageType.INFORMATION, "connectionString= "+callConfiguration.connectionString );
+        Logger.logMessage(Logger.MessageType.INFORMATION, "sourceIdentity= "+callConfiguration.sourceIdentity );
+        Logger.logMessage(Logger.MessageType.INFORMATION, "sourcePhoneNumber= "+callConfiguration.sourcePhoneNumber );
```
Comment (v-mausha) [inline CallPlayAudio/src/main/java/com/communication/CallPlayAudio/App.java]: These logs are not required.

File: CallPlayAudio/src/main/java/com/communication/CallPlayAudio/CallPlayTerminate.java (hunk 1 above)
Comment (v-maku23) [reply CallPlayAudio/src/main/java/com/communication/CallPlayAudio/CallPlayTerminate.java:11]: removed many not required imports

File: CallPlayAudio/src/main/java/com/communication/CallPlayAudio/CallPlayTerminate.java (hunk 2 above)
Comment (v-maku23) [reply CallPlayAudio/src/main/java/com/communication/CallPlayAudio/CallPlayTerminate.java:40]: removed not required variables

File: CallPlayAudio/src/main/java/com/communication/CallPlayAudio/CallPlayTerminate.java (hunk 4 above)
Comment (v-maku23) [reply CallPlayAudio/src/main/java/com/communication/CallPlayAudio/CallPlayTerminate.java]: removed not required methods

File: CallPlayAudio/src/main/java/com/communication/CallPlayAudio/App.java (hunk 5 above)
Comment (v-maku23) [reply CallPlayAudio/src/main/java/com/communication/CallPlayAudio/App.java:98]: corrected as its not pair now

File: CallPlayAudio/src/main/java/com/communication/CallPlayAudio/App.java (hunk 6 above)
Comment (v-maku23) [reply CallPlayAudio/src/main/java/com/communication/CallPlayAudio/App.java]: removed these logs

File: CallPlayAudio/pom.xml [hunk 7]
```diff
@@ -0,0 +1,158 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
```
Comment (moirf) [inline CallPlayAudio/pom.xml:5]: If this is required? and its others reference.
Please cleanup which might not required.

Comment (v-maku23) [reply CallPlayAudio/pom.xml:5]: removed it, not able to remove others

File: CallPlayAudio/src/main/java/com/communication/CallPlayAudio/CallPlayTerminate.java (hunk 3 above)
Comment (v-maku23) [reply CallPlayAudio/src/main/java/com/communication/CallPlayAudio/CallPlayTerminate.java]: removed commented ones

**Pull request merged**